### PR TITLE
MAGECLOUD-4164  [Spike] Investigate issue with elasticsearch image on AWS

### PR DESCRIPTION
**Pull request by arhiopterecs**
### Description
Added option `--es-env-var` for `docker-compose.yml` file generator. Using this option, you can specify the environment variables that will be applied when starting the Elasticsearch container.
This will allow you to change service settings, such as heap size for JVM or Elasticsearch options.
Parameter changes for JVM are available from image version 1.7
Changing Elasticsearch parameters using environment variables is available starting from version 5.2

### Fixed Issues (if relevant)
1. https://magento2.atlassian.net/browse/MAGECLOUD-4164

### Manual testing scenarios
1. Clone `[magento-cloud template](https://github.com/magento/magento-cloud)`.
2. Run: `$ composer update` in the magento-cloud template directory: 
3.  Generate the docker-compose.yml file with env variables for `elasticsearch` service: 
```bash
$ php vendor/bin/ece-docker build:compose --es-env-var=ES_JAVA_OPTS="-Xms512m -Xmx512m" --es-env-var=node.store.allow_mmapfs=false
Configuration was built.
```
NOTE: By default, an image with Elasticsearch version 6.5 is used. For this version of Elasticsearch, the option `node.store.allow_mmapfs` exists.
For image varsions 5.2, 2.4, 1.7 ,this option does not exist
Forimage varsions 7.5, this parameter was renamed to node.store.allow_mmap
4. Check the file `docker-compose.yml`, the following settings must exist:
```
... 
elasticsearch:
    hostname: elasticsearch.magento2.docker
    image: 'magento/magento-cloud-docker-elasticsearch:6.5-1.1'
    environment:
      - 'ES_JAVA_OPTS=-Xms512m -Xmx512m'
      - node.store.allow_mmapfs=false
    networks:
      magento:
        aliases:
          - elasticsearch.magento2.docker           
....  
```
5. Run: `$ docker-compose up -d`. Containers started successfully
6. Login in elasticsearch container:
```
$ docker-compose exec elasticsearch bash
```
7. Run  the command in the container: `$ ps aux  | grep -E "(Xms|Xmx)"`
Expected: The last values ​​of the `-Xms` and` -Xmx` parameters are 512m

8. Run  the command in the container: ` $ curl http://localhost:9200/_nodes 2>&1 | grep 'mmapfs'`.
Expected:`"allow_mmapfs":"false"` should be present

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: src/Command/BuildCompose.php [hunk 1]
```diff
@@ -204,6 +204,12 @@ protected function configure(): void
                 null,
                 InputOption::VALUE_OPTIONAL,
                 'Cloud environment variables'
+            )
+            ->addOption(
+                Source\CliSource::OPTION_ES_EVN_VAR,
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                "Environment variable for elasticsearch service"
```
Comment (NadiyaS) [inline src/Command/BuildCompose.php]: It would be good to add example for this option.

**Review by NadiyaS — DISMISSED** [review]

**Review by NadiyaS — DISMISSED** [review]

File: src/Command/BuildCompose.php [hunk 2]
```diff
@@ -216,6 +216,12 @@ protected function configure(): void
                 null,
                 InputOption::VALUE_OPTIONAL,
                 'Port'
+            )
+            ->addOption(
+                Source\CliSource::OPTION_ES_EVN_VAR,
+                null,
+                InputOption::VALUE_OPTIONAL | InputOption::VALUE_IS_ARRAY,
+                "Environment variable for elasticsearch service"
```
Comment (shiftedreality) [inline src/Command/BuildCompose.php]: Please use single quotes

Comment (shiftedreality) [reply src/Command/BuildCompose.php]: I believe it's also optional by default

File: src/Config/Source/CliSource.php [hunk 3]
```diff
@@ -54,6 +54,11 @@ class CliSource implements SourceInterface
     public const OPTION_HOST = 'host';
     public const OPTION_PORT = 'port';
 
+    /**
+     * Environment variable for elasticsearch service.
+     */
+    public const OPTION_ES_EVN_VAR = 'es-env-var';
```
Comment (shiftedreality) [inline src/Config/Source/CliSource.php]: EVN => ENV

Comment (shiftedreality) [reply src/Config/Source/CliSource.php]: Please use full name, `OPTION_ES_ENVIRONMENT_VARIABLE`

File: src/Config/Source/SourceInterface.php [hunk 4]
```diff
@@ -63,6 +63,11 @@ interface SourceInterface
      */
     public const SERVICES_ES = self::SERVICES . '.' . ServiceInterface::SERVICE_ELASTICSEARCH;
 
+    /**
+     * ES environment variables
+     */
+    public const SERVICES_ES_ENV_VARS = self::SERVICES_ES.'.'.'env-vars';
```
Comment (shiftedreality) [inline src/Config/Source/SourceInterface.php]: I would also suggest dropping `env` part, variables are already self-explaining

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/Config/Source/SourceInterface.php (hunk 4 above)
Comment (arhiopterecs) [reply src/Config/Source/SourceInterface.php]: Done

File: src/Config/Source/CliSource.php (hunk 3 above)
Comment (arhiopterecs) [reply src/Config/Source/CliSource.php]: Done

File: src/Command/BuildCompose.php (hunk 2 above)
Comment (arhiopterecs) [reply src/Command/BuildCompose.php]: Done

**Comment by arhiopterecs** [comment]
PR was recreated https://github.com/magento/magento-cloud-docker/pull/174

**Pull request closed without merging**